Tests/sbgTests/FoundationFileAdderTests.swift

```diff
@@ -0,0 +1,77 @@
+//
+//  FoundationFileAdderTests.swift
+//  SBGTests
+//
+//  Created by Dawid Markowski on 15/10/2018.
+//
+
+
+import Foundation
+import Quick
+import Nimble
+@testable import SBGCore
+
+class FoundationFileAdderTests: QuickSpec {
+    
+    override func spec() {
+        describe("FoundationFileAdder") {
+            
+            var sut: FoundationFileAdder!
+            var pathResolver: MockPathResolver!
+            
+            beforeEach {
+                pathResolver = MockPathResolver()
+                sut = FoundationFileAdder(pathResolver: pathResolver)
+            }
+            
+            context("when adding file") {
+                beforeEach {
+                    try! sut.addFile(with: MockConstants.sampleName, content: MockConstants.sampleContent, to: MockConstants.sampleDirectory)
+                }
+                
+                it("should invoke path resolver excactly once") {
+                    expect(pathResolver.invocationCount).to(equal(1))
+                }
+                
+                it("should invoke path resolver with correct file name") {
+                    expect(pathResolver.name).to(equal(MockConstants.sampleName))
```

Comment:
`expect(pathResolver.name) == MockConstants.sampleName` (the same applies to the others)